Docker

KirillKotikov opened the pull request:
[ + ] Добавил модуль с использованием докера.

Commit:
Docker
[ + ] Добавил модуль с использованием докера.

vlad-iv commented on util/docker-library/docker-compose.yml:
```diff
@@ -0,0 +1,36 @@
+      - SPRING_DATASOURCE_PASSWORD=postgres
+    volumes:
+      - ./logs:/var/log
+      - ./data:/service
```
предлагаю удалить

vlad-iv commented on util/docker-library/Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM khipu/openjdk17-alpine
+WORKDIR /service
+COPY docker-library-1.0.jar /docker-library.jar
```
После сборки в корне нет артефакта
предлагаю поправить путь на target/....

vlad-iv commented on util/docker-library/docker-compose.yml:
```diff
@@ -0,0 +1,36 @@
+  library:
+    ports:
+      - "8080:8080"
+    build:
```
предлагаю указать depends_on:
так как без БД приложение не работает

vlad-iv commented on util/docker-library/docker-compose.yml:
```diff
@@ -0,0 +1,36 @@
+    image: library-image:1.0
+    restart: always
+    container_name: library
+    environment:
```
предлагаю задать профайл prod (ведь создан файл application-prod.yml ? )

Commit:
Merge branch 'main' of https://github.com/KirillKotikov/2023-03-otus-spring-kotikov into hw18-docker

# Conflicts:
#	README.md
#	util/README.md
#	util/pom.xml

Commit:
Docker
[ - ] Удалил ненужное из docker-compose.yml.
[ + ] Добавил переменную PROFILE в docker-compose.yml.
[ * ] Изменил Dockerfile, чтобы сборка работала из корня модуля.
[ + ] Добавил переменную среды для обозначения profile в application.yml.
